src/org/opensolaris/opengrok/web/PageConfig.java

```diff
@@ -391,8 +392,22 @@ public String canProcess() {
                 dirFileList = Collections.emptyList();
             } else {
                 Arrays.sort(files, String.CASE_INSENSITIVE_ORDER);
-                dirFileList
-                        = Collections.unmodifiableList(Arrays.asList(files));
+                List<String> list = Arrays.asList(files);
+                if (getPath().isEmpty()) {
+                    /**
+                     * This denotes the source root directory, we need to filter
+                     * projects which aren't allowed by the authorization. E. g.
+                     * remove all which aren't among the filtered set of
+                     * projects.
+                     */
+                    list = new ArrayList<>(list);
```

Comment:
I'd rather have the `list` variable named differently than the original, will make it easier to debug.